addons/hr_holidays/models/res_company.py

```diff
@@ -0,0 +1,132 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import warnings
+import pytz
+from datetime import datetime, time
+
+from odoo import models
+
+warnings.filterwarnings("ignore", category=DeprecationWarning)
+import holidays  # noqa: E402
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    def _convert_to_utc(self, holiday_date, user_tz):
+        holiday_start_utc = user_tz.localize(datetime.combine(holiday_date, time.min)).astimezone(pytz.utc).replace(tzinfo=None)
+        holiday_end_utc = user_tz.localize(datetime.combine(holiday_date, time.max)).astimezone(pytz.utc).replace(tzinfo=None)
+        return holiday_start_utc, holiday_end_utc
```

Comment:
I'm wondering if the timezone used to localize the public holidays shouldn't be adapted to something more related to the company (timezone of the resource calendar?) as a user could create the leaves for another country, or even the cron might imply wrong timezones